terraform/network.tf

```diff
@@ -0,0 +1,20 @@
+locals {
+  # Allocated per-region subnets, keyed by region
+  gcp_region_subnets = {
+    # Start at 2 to reserve some subnets for special uses if needed
+    "us-west1" = cidrsubnet("10.0.0.0/8", 3, 2)
```

Comment:
I don't understand why we need this map. Does every env running in us-west1 share 10/8?

Reply:
Sorry, I should've probably explained this better in the PR. This is for if we want to run in more than one region later on:

Each environment in the project gets it's own VPC (being created in this file) from virtue of being a completely separate TF state/etc. VPCs are global resources and aren't associated with any particular location. They're also entirely separated networks, so, say, 10.0.0.0/8 in each VPC is subnet independent and they can't reach each other.

*Subnets* however are regional. Only hosts in the subnet's region can get addresses from it, so those need to be specific for a region. Therefore, inside the environment's VPC, I need to carve out blocks for at least each region we're running in, so that subnets can be created for each of them (in gke/network.tf).

However, in this case, our TF code actually only supports one region due to the way it's set up right now, but this is something I want to fix. (I'd started to do it for this PR, but it ended up being too much churn, so I split it off.) If we have more than one region, then it's probably quite likely that we'll want the regions to be able to see and communicate with each other, so they have to be on the same VPC. Therefore, it's not possible (at least cleanly) for us to use entirely separate tfvars/state storage for each region. At the same time, the multiple regions (when it's viable to support them in our TF setup) need to use separate address blocks so they don't conflict with each other, and so there needs to be a table of region -> address block somewhere.

---

I don't love the `3, 2` either. [They are, respectively](https://www.terraform.io/docs/configuration/functions/cidrsubnet.html) `newbits` and `netnum`. So the 3 means we want to turn the `/8` into a `/11` (which means we get 2^3=8 divisions of our `/8`), and, of those, choose the 2nd subnet. If we had multiple entries in the table it'd look something like this:

```
gcp_region_subnets = {
  # Reserved: 10.0.0.0/11
  # Reserved: 10.32.0.0/11
  "us-west1" = cidrsubnet("10.0.0.0/8", 3, 2) # 10.64.0.0/11
  "us-east1" = cidrsubnet("10.0.0.0/8", 3, 3) # 10.96.0.0/11
  "europe-west1" = cidrsubnet("10.0.0.0/8", 3, 4) # 10.128.0.0/11
  # etc.
  "europe-west2" = cidrsubnet("10.0.0.0/8", 3, 7) # 10.224.0.0/11
  # Now we've used up the entire address space.
}
```

I *think* there was a reason I didn't also use the `cidr` module like I did in gke/network.tf, but I can't remember what it was anymore. I'll update the change to use it and push a new version if there's no issue, since I think it should be a bit clearer than this.

Reply:
> This is for if we want to run in more than one region later on
I think we shouldn't try to solve that problem (and take on the burden of testing and maintaining the solution) until we have that problem. In the meantime we should aim to keep everything as simple as possible.